pkg/metrics_pusher/pusher.go

```diff
@@ -58,7 +61,22 @@ func (p *MetricsPusher) pushMetrics() {
 			return
 		}
 	} else {
-		resp, err := http.Get(p.SourceURL)
+		caCert, err := ioutil.ReadFile(p.Clientca)
+		if err != nil {
+			p.Log.Error(err, "Unable to read CA cert for fetching metrics: Please provide valid CA cert or path")
+			return
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{RootCAs: caCertPool},
```

Comment:
We should set TLSHandshakeTimeout to 10 seconds otherwise there is no timeout.  As for setting other options, I'll defer to Cesar.

Reply:
Updated, and waiting for Cesar if any other options needed. Once he confirmed i will push the **Add timeout** changes.

Reply:
@pradeepnaramaneni5 it's useful to invoke SetTransportDefaults() on your transport so you get other default timeouts used in kubernetes as well.
package is `k8s.io/api-machinery/pkg/util/net`
Just invoke:
`net.SetTransportDefaults(tr)`

Reply:
@rtheis  / @csrwng Added the defaults packages.